### DISCUSSION TITLE
set OC to save only specific layers

### DISCUSSION THREAD

**Discussion by RdWitte** [modflowpy/flopy#2104] [Q&A] [answered]
Im currently trying to run a transient MF6 simulation where I want to save only head data from specific layers, for example only layer 1 and 2 (the model has 34 layers) to the .HDS file.

I'm using flopy.mf6.ModflowGwfoc to set the OC-package and have tried to set the output to "Head 1 2" to save only layer 1 and 2. 

However, this results in a unsuccesful run, generating the follow error report:
_ERROR REPORT:

  1.  ERROR READING OUTPUT CONTROL PERIOD BLOCK:
  2. UNRECOGNIZED KEYWORD: HEAD 1 2_

Does anyone have an idea on how to go about this? Thanks in advance!

The oc-package: 
 
  _package_name = oc
filename = GWM_Haarlem.oc
package_type = oc
model_or_simulation_package = model
model_name = GWM_Haarlem

Block options
--------------------
budget_filerecord
{internal}
(rec.array([('GWM_Haarlem.cbc',)],
          dtype=[('budgetfile', 'O')]))

head_filerecord
{internal}
(rec.array([('GWM_Haarlem.hds',)],
          dtype=[('headfile', 'O')]))

Block period
--------------------
saverecord
{internal}
(rec.array([('HEAD 1 2', 'LAST', None), ('BUDGET', 'LAST', None)],
          dtype=[('rtype', 'O'), ('ocsetting', 'O'), ('ocsetting_data', 'O')]))

printrecord
None_

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @RdWitte,

This is unfortunately possible in the OC package. From the [MF6 input/output guide](https://water.usgs.gov/water-resources/software/MODFLOW-6/mf6io_6.4.4.pdf): 

> For the PRINT and SAVE options of heads, there is no longer an option to specify individual layers. Whenever one
of these arrays is printed or saved, all layers are printed or saved

So the text must be "HEAD" and cannot contain any layer information. 

A workaround could be to create an Observations package that saves the heads for the cells of the layers you are interested in. For more information see the section on "Observation (OBS) Utility". This observation package can be added to your model with `flopy.mf6.ModflowUtlobs`. [This example](https://flopy.readthedocs.io/en/latest/Notebooks/mf6_complex_model_example.html) in the flopy docs also shows how to create one of these packages (in cell 11). This package allows you to save heads per cell and time step in a text file (CSV) or a binary file (BSV).

I hope this helps! Cheers,

Davíd

**Reply by RdWitte** [reply]
Hi Davíd,

Thanks for your response, and suggestion for the workaround. Will try to make do with the OBS utility package.

Riemer
